database/gormdb_assignment.go

```diff
@@ -180,31 +180,31 @@ func (db *GormDB) updateGradingCriteria(tx *gorm.DB, assignment *qf.Assignment)
 	return nil
 }
 
-// GetAssignmentsWithSubmissions returns all course assignments
-// preloaded with submissions of the requested submission type.
-func (db *GormDB) GetAssignmentsWithSubmissions(req *qf.SubmissionRequest) ([]*qf.Assignment, error) {
-	var assignments []*qf.Assignment
-	m := db.conn.Preload("Submissions").
-		Preload("Submissions.Reviews").
-		Preload("Submissions.Reviews.GradingBenchmarks").
-		Preload("Submissions.Reviews.GradingBenchmarks.Criteria").
-		Preload("Submissions.Scores")
+// GetCourseSubmissions returns the latest course submissions of the requested submission type.
+func (db *GormDB) GetCourseSubmissions(courseID uint64, submissionType qf.SubmissionRequest_SubmissionType) ([]*qf.Submission, error) {
+	var assignmentIDs []uint64
+	a := db.conn.Model(&qf.Assignment{}).Where("course_id = ?", courseID)
```

Comment:
Would it be better to it like this (to be type safe)?:
```go
a := db.conn.Model(&qf.Assignment{}).Where(&qf.Assignment{CourseID: courseID})
switch submissionType {
case qf.SubmissionRequest_USER:
	a.Where(&qf.Assignment{IsGroupLab: false})
case qf.SubmissionRequest_GROUP:
	a.Where(&qf.Assignment{IsGroupLab: true})
default: // all
}
```
That is, is there a good reason to use the string-based versions?